homeassistant/components/subaru/sensor.py

```diff
@@ -215,13 +215,13 @@ def native_value(self) -> None | int | float:
         if current_value is None:
             return None
 
-        if unit in LENGTH_UNITS:
+        if unit in LENGTH_UNITS and unit_system == METRIC_SYSTEM:
             return round(unit_system.length(current_value, cast(str, unit)), 1)
 
-        if unit in PRESSURE_UNITS and unit_system == US_CUSTOMARY_SYSTEM:
+        if unit in PRESSURE_UNITS and unit_system == METRIC_SYSTEM:
             return round(
-                unit_system.pressure(current_value, cast(str, unit)),
-                1,
+                PressureConverter.convert(current_value, unit, UnitOfPressure.KPA),
+                0,
```

Comment:
Why are we manually converting units here?

As in, shouldn't we just set the suggest unit of measurement and let Home Assistant handle the conversion instead?

Reply:
Updated to now let Home Assistant handle all sensor unit conversions (except for average fuel consumption)